nuxeo-services/nuxeo-invite/src/main/java/org/nuxeo/ecm/user/invite/DefaultInvitationUserFactory.java

```diff
@@ -79,4 +84,36 @@ public NuxeoPrincipal doCreateUser(CoreSession session, DocumentModel registrati
         }
         return user;
     }
+
+    /**
+     * Filter group by computing the intersection of the group in the registration doc and the groups of the user that
+     * created the request. Administrators accept all groups.
+     *
+     * @param registrationDoc
+     * @param configuration
+     *
+     * @since 10.2
+     */
+    private List<String> filterGroups(DocumentModel registrationDoc, UserRegistrationConfiguration configuration) {
+        List<String> wantedGroup = (List<String>) registrationDoc.getPropertyValue(
+                configuration.getUserInfoGroupsField());
+
+        String originatingUser = registrationDoc.getProperty(UserInvitationComponent.PARAM_ORIGINATING_USER)
+                                                .getValue(String.class);
+        UserManager userManager = Framework.getService(UserManager.class);
+        NuxeoPrincipal originatingPrincipal = userManager.getPrincipal(originatingUser);
```

Comment:
Add a null check, if the user doesn't exist (has been removed since, for instance).